src/Cli/dotnet/NugetSearch/NugetToolSearchApiRequest.cs

```diff
@@ -42,10 +44,10 @@ public async Task<string> GetResult(NugetSearchApiParameter nugetSearchApiParame
             return await response.Content.ReadAsStringAsync(cancellation.Token);
         }
 
-        internal static Uri ConstructUrl(string searchTerm = null, int? skip = null, int? take = null,
-            bool prerelease = false)
+        internal static async Task<Uri> ConstructUrl(string searchTerm = null, int? skip = null, int? take = null,
+            bool prerelease = false, Uri domainAndPathOverride = null)
         {
-            var uriBuilder = new UriBuilder("https://azuresearch-usnc.nuget.org/query");
+            var uriBuilder = new UriBuilder(domainAndPathOverride ?? await DomainAndPath());
```

Comment:
To be clear, this `domainAndPathOverride` is just for unit testing, right? So things can be tested offline?

Reply:
yes, for unit testing